### PR TITLE
core: fix promise microtask running sequence

### LINKED ISSUE #107: Issue on Async Running Sequence
It seems that txiki's event loop doesn't follow the microtask / macrotask concepts strictly. For example:

``` js
console.log('Hello')

setTimeout(() => console.log('!'), 0)

Promise.resolve().then(() => console.log('World'))
```

This snippet has following output in Chrome's console and `qjs`:

```
Hello
World
!
```

Since `Promise.then` returns a microtask, and if my understanding is correct, it should be executed by `JS_ExecutePendingJob` synchronously right after each eval. So the log from `Promise.then` comes before the one from `setTimeout`.

While in txiki I got following output:

```
Hello
!
World
```

Currently microtasks are triggered by `uv_check` after each tick, while after the first time we eval the script, there isn't corresponding `JS_ExecutePendingJob` calls, so the `Promise`'s microtask is kept until next tick's check phase in `setTimeout`'s macrotask, which makes the difference in running sequence.

### PR DESCRIPTION
**Pull request by doodlewind**
Fix #107 

We don't have to migrate to `UV_RUN_ONCE` for this case. This PR simply executes the microtask jobs before the `setTimeout` callback, which makes the `Promise.then` microtasks can be always cleared before `setTimeout`.

I've verified the basic and the nesting callback case in the original issue.

### CODE REVIEW

**Comment by saghul** [comment]
Hum, interesting way of solving it! Can you please DRY the execute_jobs function and expose it in private.h so we can use it in both places we need it?

**Comment by saghul** [comment]
Also, if you could add a test that would be ideal.

File: src/vm.c [hunk 1]
```diff
@@ -258,23 +258,26 @@ static void uv__prepare_cb(uv_prepare_t *handle) {
...
-    JSRuntime *rt = qrt->rt;
-    JSContext *ctx1;
+void tjs_execute_jobs(JSContext *ctx) {
+    JSRuntime *rt = JS_GetRuntime(ctx);
     int err;
 
     /* execute the pending jobs */
     for (;;) {
-        err = JS_ExecutePendingJob(rt, &ctx1);
+        err = JS_ExecutePendingJob(rt, &ctx);
```
Comment (saghul) [inline src/vm.c]: THis is not correct. Please leave ctx1 and take a TJSRuntime as an argument to tjs_execute_jobs. The reason is that JS_ExecutePendingJob returns the context that is resumed in the passed ctx argument. It need not be the one we have (if we ever go multi-context).

Comment (doodlewind) [reply src/vm.c]: I can't get the `TJSRuntime` directly in the timer's callback. Current solution aligns with the `js_std_loop` pattern in QuickJS.

**Comment by doodlewind** [comment]
@saghul timer tests are also ready 😀

**Comment by saghul** [comment]
@doodlewind Awesome work, thank you! 🚀

**Pull request merged**